contracts/tests/conftest.py

```diff
@@ -153,3 +159,36 @@ def block_header_by_malicious_validator(malicious_validator_key):
 @pytest.fixture()
 def block_header_by_malicious_non_validator(malicious_non_validator_key):
     return make_block_header(private_key=malicious_non_validator_key)
+
+
+@pytest.fixture(scope="session")
+def no_whitelist_validator_auction_contract(deploy_contract):
+    contract = deploy_contract("TestValidatorAuctionFixedPrice")
+    return contract
+
+
+@pytest.fixture(scope="session")
+def validator_auction_contract(
+        no_whitelist_validator_auction_contract,
+        extended_accounts,
+        send_ether_to_extended_accounts,
+):
+    no_whitelist_validator_auction_contract.functions.addToWhitelist(extended_accounts[1:]).transact()
+    return no_whitelist_validator_auction_contract
+
+
+@pytest.fixture(scope="session")
+def send_ether_to_extended_accounts(chain, extended_accounts, number_of_auction_participants):
+    account_0 = chain.get_accounts()[0]
+
+    for i in range(1, 1 + number_of_auction_participants):
+        chain.send_transaction({'from': account_0, 'to': extended_accounts[i], 'gas': 21000, 'value': 10000000})
+
+
+@pytest.fixture(scope="session")
+def extended_accounts(chain, number_of_auction_participants):
+    new_chain = chain
+    for i in range(100, 100 + number_of_auction_participants):
+        new_chain.add_account("0x0000000000000000000000000000000000000000000000000000000000000" + str(i))
+
+    return new_chain.get_accounts()
```

Comment:
This should be changed to return only the possible participants. At the moment the rest of the code is littered with
```
for i in range(1, ...):
   do_something_with(extended_account[i])
 ```
It would be much nicer if we could iterate over the list directly instead of iterating over the indexes. It would become something like
```
for participant_account in extended_accounts:
  do_something_with(participant_account)
```
Also, I suggest using a better name. `participant_accounts` would be much clearer to me. 
And the last thing is,we generate exactly `number_of_auction_participants` accounts,but we will have more whitelisted accounts than participants. I think we should whitelist more accounts in the test setup.